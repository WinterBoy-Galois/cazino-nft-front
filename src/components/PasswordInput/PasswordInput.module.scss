@import '../../styles/shared';

.inputField {
  text-align: left;
  background-color: $form-background;
  color: $white;
  font-size: 18px;
  line-height: 24px;
  margin-left: 0;
  padding: 0;
  outline: none;
  border: none;

  &__container {
    min-width: 250px;
    width: 100%;
    margin-bottom: 1rem;
  }

  &__wrapper {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: row;
    align-items: center;
    padding: 5px 10px 5px 10px;
    border: 2px solid $form-background;
    border-radius: 8px;
    background-color: $form-background;
    height: 57px;
    vertical-align: bottom;
    transition: border 0.2s ease-in;

    &:focus-within {
      border: 2px solid $bitcoinYellow;
    }

    &--error {
      border: 2px solid $red;

      &:focus-within {
        border: 2px solid $red;
      }
    }

    input[type='text'].inputField,
    input[type='password'].inputField {
      width: 100%;
      min-width: 0;
    }

    input {
      filter: none;
    }

    input:-webkit-autofill,
    input:-webkit-autofill:hover,
    input:-webkit-autofill:focus {
      border: none;
      -webkit-text-fill-color: $white;
      -webkit-box-shadow: 0 0 0 1000px $form-background inset;
      box-shadow: 0 0 0 1000px $form-background inset;
      transition: background-color 5000s ease-in-out 0s;
    }
  }

  &__group {
    flex-direction: column;
    align-items: flex-start;
    flex-grow: 1;
  }

  &__icon {
    cursor: pointer;
    width: 25px;
    margin-left: 10px;
    border: none;
    padding: 0;
    background: none;

    &--visible {
      stroke: $eye-outline;
      transition: stroke 0.2s ease-in;

      &:hover {
        stroke: $live-updates-light-blue;
      }
    }

    &--invisible {
      fill: $eye-outline;
      transition: fill 0.2s ease-in;

      &:hover {
        fill: $live-updates-light-blue;
      }
    }
  }

  &__label {
    text-transform: uppercase;
    display: block;
    font-size: 14px;
    line-height: 19px;
    color: $live-updates-light-blue;
    user-select: none;
    margin: 0;
  }

  &__error {
    text-transform: uppercase;
    display: block;
    font-size: 14px;
    line-height: 19px;
    color: $red;
    user-select: none;
    margin: 4px 8px 4px 8px;
  }
}
