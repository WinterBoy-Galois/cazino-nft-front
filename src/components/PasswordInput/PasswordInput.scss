.inputField__container {
  min-width: 250px;
  width: 100%;
  display: flex;
}

.inputField__wrapper {
  flex: 1;
  display: inline-block;
  padding: 5px 10px 5px 10px;
  border: 4px solid #0a1a33;
  border-radius: 8px;
  background-color: #0a1a33;
  height: 52px + (2 * 8px);
  vertical-align: bottom;
  transition: all 0.2s ease-in;
}

.inputField__wrapper:hover {
  border: 4px solid #fec41c;
}

.inputField {
  text-align: left;
  background-color: #0a1a33;
  color: #fff;
  font-weight: 300;
  font-size: 18px;
  line-height: 24px;
  margin-left: 0;
  padding: 0;
  // box-sizing: border-box;
  transition: all 0.2s ease-in;
  outline: none;
  border: none;
  // width: auto;
  // display: block;
  // user-select: none;
}

input[type='text'].inputField {
  size: '';
  width: 100%;
}

input[type='text'].inputField:focus {
  background-color: #133252;
  border-color: #275e9a;
}

.inputFieldLabel {
  text-transform: uppercase;
  display: block;
  font-size: 14px;
  font-weight: 300;
  line-height: 19px;
  color: #80a2cc;
  user-select: none;
  margin: 0;
}
