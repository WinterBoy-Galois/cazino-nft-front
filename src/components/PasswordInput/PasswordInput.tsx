import React, { useState, useEffect } from 'react';

import styles from './PasswordInput.module.scss';
import { Eye, EyeInvisible } from '..';

interface IProps {
  name?: string;
  value?: string;
  label?: string;
  validationMessage?: string;
  onChangeValue?: (value: string) => void;
  onBlur?: ({ target }: { target: EventTarget | null }) => void;
}

const PasswordInput = ({
  name,
  value: initialValue,
  label,
  validationMessage,
  onChangeValue,
  onBlur,
}: IProps) => {
  const [value, setValue] = useState(initialValue);
  const [isPasswordVisible, setIsPasswordVisible] = useState(false);

  useEffect(() => {
    setValue(initialValue);
  }, [initialValue]);

  const isError = () => {
    return !!validationMessage;
  };

  const handleOnChange = (e: { target: { value: string } }) => {
    const newValue = e.target.value;

    setValue(newValue);
    if (onChangeValue) {
      onChangeValue(newValue);
    }
  };

  const handleBlur = (event?: React.FocusEvent<HTMLInputElement>) => {
    if (onBlur && event) {
      onBlur(event);
    }
  };

  const handleClick = () => {
    setIsPasswordVisible(!isPasswordVisible);
  };

  return (
    <div className={styles.inputField__container}>
      <div
        className={
          !isError()
            ? `${styles['inputField__wrapper']}`
            : `${styles['inputField__wrapper']} ${styles['inputField__wrapper--error']}`
        }
      >
        <div className={styles.inputField__group}>
          <label className={styles.inputField__label}>{label}</label>
          <input
            {...(name ? { name: name } : {})}
            type={isPasswordVisible ? 'text' : 'password'}
            value={value}
            className={styles.inputField}
            autoComplete="off"
            onChange={handleOnChange}
            onBlur={handleBlur}
          />
        </div>
        <button
          type="button"
          className={`${styles['inputField__icon']}`}
          onClick={handleClick}
          tabIndex={-1}
        >
          {isPasswordVisible ? (
            <Eye className={`${styles['inputField__icon--visible']}`} />
          ) : (
            <EyeInvisible className={`${styles['inputField__icon--invisible']}`} />
          )}
        </button>
      </div>
      {isError() && <div className={styles.inputField__error}>{validationMessage}</div>}
    </div>
  );
};

export default PasswordInput;
