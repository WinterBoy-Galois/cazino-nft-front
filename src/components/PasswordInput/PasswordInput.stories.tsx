import React from 'react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text } from '@storybook/addon-knobs';

import PasswordInput from '.';

export default {
  title: 'Components/Forms/PasswordInput',
  component: PasswordInput,
  decorators: [
    withKnobs,
    (storyFn: () => React.ReactNode) => <div style={{ padding: '1rem' }}>{storyFn()}</div>,
  ],
  parameters: {
    backgrounds: {
      default: 'Secondary',
    },
  },
};

const data = {
  label: 'Password',
  value: 'Password123',
  onChangeValue: action('onChangeValue'),
};

export const Default = () => (
  <PasswordInput {...data} label={text('Label', data.label)} value={text('Value', data.value)} />
);
