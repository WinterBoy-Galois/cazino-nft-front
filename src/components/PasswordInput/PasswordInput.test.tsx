import React from 'react';
import { render } from '@testing-library/react';

import PasswordInput from '.';

const data = {
  label: 'Password',
  value: 'Password123',
};

describe('PasswordInput', () => {
  it('should match snapshot', async () => {
    // Arrange

    // Act
    const { container } = render(<PasswordInput {...data} />);

    // Assert
    expect(container).toMatchSnapshot();
  });

  it('should set value', () => {
    // Arrange
    const expected = data.label;

    // Act
    const { getByDisplayValue } = render(<PasswordInput value={expected} />);
    const container = getByDisplayValue(expected);

    // Assert
    expect(container).toHaveAttribute('value', expected);
  });

  it('should set name', () => {
    // Arrange
    const expected = data.value;

    // Act
    const { getByDisplayValue } = render(<PasswordInput value={data.label} name={expected} />);
    const container = getByDisplayValue(data.label);

    // Assert
    expect(container).toHaveAttribute('name', expected);
  });
});
