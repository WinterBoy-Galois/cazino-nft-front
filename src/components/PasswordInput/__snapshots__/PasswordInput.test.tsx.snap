// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInput should match snapshot 1`] = `
<div>
  <div
    class="inputField__container"
  >
    <div
      class="inputField__wrapper"
    >
      <div
        class="inputField__group"
      >
        <label
          class="inputField__label"
        >
          Password
        </label>
        <input
          autocomplete="off"
          class="inputField"
          type="password"
          value="Password123"
        />
      </div>
      <button
        class="inputField__icon"
        tabindex="-1"
        type="button"
      >
        <svg
          class="fill inputField__icon--invisible"
          viewBox="0 0 28.49 21.53"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
            id="Layer_2"
          >
            <g
              data-name="Layer 1"
              id="Layer_1-2"
            >
              <path
                d="M28.33 10.1a22.8 22.8 0 00-6.44-7.51l-1.72 2.47a19.79 19.79 0 015.1 5.71c-1.2 2-5.15 7.76-11 7.76a9.26 9.26 0 01-3.08-.53l-1.76 2.54a12.26 12.26 0 004.81 1c8.94 0 13.88-9.68 14.09-10.09a1.5 1.5 0 000-1.35zM8.31 16.47a19.72 19.72 0 01-5.09-5.7c1.2-2 5.15-7.77 11-7.77a9.27 9.27 0 013.06.53L19.06 1a12.31 12.31 0 00-4.82-1C5.31 0 .36 9.68.16 10.1a1.5 1.5 0 000 1.34 22.64 22.64 0 006.44 7.5z"
              />
              <path
                d="M12.27 10.77a2 2 0 011.28-1.85l2-2.92a4.88 4.88 0 00-1.34-.21 5 5 0 00-4 7.9zM16.22 10.76a2 2 0 01-1.29 1.84l-2 2.92a4.63 4.63 0 001.34.21 5 5 0 005-5 5 5 0 00-1-2.93zM7.48 21.38a1 1 0 01-.48-.16.94.94 0 01-.23-1.3L20.27.45a.92.92 0 011.29-.24.93.93 0 01.23 1.3L8.25 21a.93.93 0 01-.77.38z"
              />
            </g>
          </g>
        </svg>
      </button>
    </div>
  </div>
</div>
`;
